src/wilcoxon.jl

```diff
@@ -163,6 +163,27 @@ function StatsAPI.pvalue(x::ExactSignedRankTest; tail=:both)
     end
 end
 
+function psignrank(W::Union{Float64, Int}, n::Int, lower_tail::Bool)
+    W = round(Int, W)
+    max_W =  (n * (n + 1)) >> 1
+    if !lower_tail
+        return psignrank(max_W - W, n, true)
+    end
+    if W == max_W
+        return 1.0
+    elseif W > max_W >> 1
+        return 1 - psignrank(max_W - W - 1, n, true)
+    end
+    DP = zeros(Int, W + 1)
+    DP[W+1] = 1
+    for j in 1:n
+        for i in 1:(W+1-j)
+            DP[i] += DP[i + j]
+        end
+    end
+    return sum(ldexp.(float.(DP), -n))
```

Comment:
We can avoid broadcasting and reduce allocations here:
```suggestion
    return sum(Base.Fix2(ldexp, -n) ∘ float, DP)
```

Reply:
Map reduce also gets rid of the allocation, not sure what the exact difference is.
```suggestion
    mapreduce(x->ldexp(float(x), -n), +, DP)
```